src/Chapter2.hs

```diff
@@ -642,8 +644,8 @@ takeEven :: [a] -> [a]
 takeEven = go ([], 0)
   where
     go :: ([a], Int) -> [a] -> [a]
-    go (res, _) [] = res
-    go (res, i) (x:xs) = if even i then go (res ++ [x], i+1) xs else go (res, i+1) xs
+    go (res, _) [] = reverse res
+    go (res, i) (x:xs) = if even i then go (x:res, i+1) xs else go (res, i+1) xs
```

Comment:
This is correct and almost perfect :ok_hand: 

Though, `reverse` is slow as it traverses the whole list (as you know because you've implemented `rewind`. Note, that here, if you replace `go (x : res)` with `x : go xs`, you will get a proper order of elements without the need to use `reverse` in the end, and you will also get rid of the additional variable `res`